components/Email.jsx

```diff
@@ -0,0 +1,102 @@
+import React, { useRef } from "react";
+import emailjs from "emailjs-com";
+
+import { ToastContainer, toast } from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
+
+const ContactForm = () => {
+  const form = useRef();
+
+  const sendEmail = (e) => {
+    e.preventDefault();
+
+    let userName = e.target.elements.user_name.value;
+    let userEmail = e.target.elements.user_email.value;
+    let subject = e.target.elements.subject.value;
+    let message = e.target.elements.message.value;
+
+    if (userName == "" || userEmail == "" || subject == "" || message == "") {
+      toast.error("Please fill out the form before hitting the send button.");
+      return;
+    }
+
+    emailjs
+      .sendForm(
+        process.env.REACT_APP_EMAILJS_SERVICE_ID,
+        process.env.REACT_APP_EMAILJS_TEMPLATE_ID,
+        e.target,
+        process.env.REACT_APP_EMAILJS_USER_ID
+      )
+      .then(
+        (result) => {
+          e.target.reset();
+          toast.success("Email sent!");
```

Comment:
I think you could use a `toast.promise` to add more visual feedback to the form: https://fkhadra.github.io/react-toastify/promise

It will show a progress toast until the email is sent/fails to send.